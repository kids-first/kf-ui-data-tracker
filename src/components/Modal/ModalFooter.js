import React from 'react';
import PropTypes from 'prop-types';
import classes from 'classnames';
import {Button} from 'kf-uikit';
/**
 * Displays the footer of the modal
 */
const ModalFooter = ({
  className,
  onCloseModal,
  onSubmit,
  cancelText,
  submitText,
  disableSubmit,
  footerButton,
}) => {
  let modalFooterClass = classes('Modal--Footer', className);
  return (
    <footer className={modalFooterClass}>
      <Button className="w-32 justify-center" onClick={onCloseModal}>
        {cancelText}
      </Button>
      {footerButton}
      <Button
        onClick={onSubmit}
        className="w-32 justify-center"
        color="primary"
        disabled={disableSubmit}
      >
        {submitText}
      </Button>
    </footer>
  );
};

ModalFooter.propTypes = {
  /** Any additional classes to be applied to the modal footer*/
  className: PropTypes.string,
  /** Text on cancel button*/
  cancelText: PropTypes.string,
  /** Text on submit button*/
  submitText: PropTypes.string,
  /** Action to close modal*/
  onCloseModal: PropTypes.func,
  /** Action to submit content*/
  onSubmit: PropTypes.func,
};

ModalFooter.defaultProps = {
  className: null,
  cancelText: 'CANCEL',
  submitText: 'SUBMIT',
};

export default ModalFooter;
