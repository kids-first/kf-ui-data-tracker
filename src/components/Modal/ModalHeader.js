import React from 'react';
import PropTypes from 'prop-types';
import classes from 'classnames';
import {Icon} from 'kf-uikit';
/**
 * Displays the footer of the modal
 */
const ModalHeader = ({className, onCloseModal, title}) => {
  let modalHeaderClass = classes('Modal--Header', className);
  return (
    <header className={modalHeaderClass}>
      <h2 className="Modal--Title">{title}</h2>
      <button
        className="mt-4"
        onClick={onCloseModal}
        data-testid="close-modal-button"
      >
        <Icon kind="close" />
      </button>
    </header>
  );
};

ModalHeader.propTypes = {
  /** Any additional classes to be applied to the modal header*/
  className: PropTypes.string,
  /** Text on modal header*/
  title: PropTypes.string,
  /** Action to close modal*/
  onCloseModal: PropTypes.func,
};

ModalHeader.defaultProps = {
  className: null,
  title: null,
};

export default ModalHeader;
