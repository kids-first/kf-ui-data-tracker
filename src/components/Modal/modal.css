.Modal--container {
  border-radius: 5px;
  position: absolute;
  top: 8%;
  bottom: 8%;
  left: 20%;
  right: 20%;
  @extend .bg-white, .flex, .flex-col, .content-between, .justify-between;
}

.Modal--overlay {
  background: rgba(0, 0, 0, 0.4);
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}

.Modal--content {
  @extend .flex-1, .overflow-y-scroll, .mx-20, .my-12;
}

.Modal--Footer {
  border-radius: 0 0 5px 5px;
  @extend .bg-lightGrey, .p-20, .flex, .justify-between, .items-center;
}

.Modal--Header {
  @extend .flex,
    .justify-between,
    .content-center,
    .border-b,
    .border-mediumGrey,
    .py-12,
    .mx-20;
}

.Modal--Title {
  @extend .inline-block, .text-blue, .my-0, .p-0;
}

.Modal--Header-Close {
  @extend .inline-block, .m-0, .p-0;
}
