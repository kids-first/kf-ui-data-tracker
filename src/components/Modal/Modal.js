import React, {Fragment} from 'react';
import PropTypes from 'prop-types';
import classes from 'classnames';
import ReactModal from 'react-modal';
import ModalHeader from './ModalHeader';
import ModalFooter from './ModalFooter';
/**
 * Displays modal view
 */
const Modal = ({
  className,
  onSubmit,
  title,
  children,
  onCloseModal,
  onBeforeClose,
  disableSubmit,
  cancelText,
  submitText,
}) => {
  let modalClass = classes('Modal--container', className);
  return (
    <ReactModal
      data-testid="modal"
      isOpen={true}
      contentLabel="File Version Uploda/Annotation Modal"
      onRequestClose={onBeforeClose}
      className={modalClass}
      overlayClassName="Modal--overlay"
      ariaHideApp={false}
    >
      <ModalHeader title={title} onCloseModal={onCloseModal} />
      <section className="Modal--content">
        <Fragment>{children}</Fragment>
      </section>
      <ModalFooter
        onCloseModal={onCloseModal}
        onSubmit={onSubmit}
        disableSubmit={disableSubmit}
        cancelText={cancelText}
        submitText={submitText}
      />
    </ReactModal>
  );
};

Modal.propTypes = {
  /** Any additional classes to be applied to the modal*/
  className: PropTypes.string,
  /** Action to submit content*/
  onSubmit: PropTypes.func,
  /** Modal title*/
  title: PropTypes.string,
};

Modal.defaultProps = {
  className: null,
};

export default Modal;
