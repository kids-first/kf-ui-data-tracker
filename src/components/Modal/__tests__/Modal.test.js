import React from 'react';
import {render, cleanup} from 'react-testing-library';
import Modal from '../Modal';

afterEach(cleanup);
it('renders modal with no content, submit is enabled', async () => {
  const {container} = render(
    <Modal
      title="MODAL TITLE"
      onSubmit={() => {}}
      onCloseModal={() => {}}
      disableSubmit={false}
    />,
    {container: document.body},
  );
  expect(container).toMatchSnapshot();
});

it('renders modal with no content, submit is disabled', async () => {
  const {container} = render(
    <Modal
      title="MODAL TITLE"
      onSubmit={() => {}}
      onCloseModal={() => {}}
      disableSubmit={true}
    />,
    {container: document.body},
  );
  expect(container).toMatchSnapshot();
});
