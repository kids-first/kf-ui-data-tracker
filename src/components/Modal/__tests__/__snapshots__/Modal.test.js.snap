// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders modal with no content, submit is disabled 1`] = `<div />`;

exports[`renders modal with no content, submit is enabled 1`] = `<div />`;
